### PR TITLE
Google Service (Drive v3 or Excel V4) added as a parameter

### PR DESCRIPTION
**Pull request by Yvelo**
# Modify GoogleDriveLoader so that it can accept a Google Service instead of relying on a path to credential files.

I am deploying LangChain in serverless environment where I use Redis for chat memory and security token store. This pull request allows to pass a Google Drive or a Google Sheet Service instead of a path to local files.

I would also consider moving _load_sheet_from_id into a different class than GoogleDriveLoader since it is not based on Google Drive API. 

Dear @eyurtsev, please advise,

### CODE REVIEW

File: langchain/document_loaders/googledrive.py [hunk 1]
```diff
@@ -264,14 +259,10 @@ def _load_file_from_id(self, id: str) -> List[Document]:
...
 
-        from googleapiclient.discovery import build
         from googleapiclient.http import MediaIoBaseDownload
 
-        creds = self._load_credentials()
-        service = build("drive", "v3", credentials=creds)
-
-        file = service.files().get(fileId=id, supportsAllDrives=True).execute()
-        request = service.files().get_media(fileId=id)
+        file = self._service().files().get(fileId=id, supportsAllDrives=True).execute()
```
Comment (hwchase17) [inline langchain/document_loaders/googledrive.py:264]: can we do `service = self._service()` and then use in both these lines?

File: langchain/document_loaders/googledrive.py [hunk 2]
```diff
@@ -170,15 +171,11 @@ def _load_document_from_id(self, id: str) -> Document:
...
-        from googleapiclient.discovery import build
         from googleapiclient.errors import HttpError
         from googleapiclient.http import MediaIoBaseDownload
 
-        creds = self._load_credentials()
-        service = build("drive", "v3", credentials=creds)
-
-        file = service.files().get(fileId=id, supportsAllDrives=True).execute()
-        request = service.files().export_media(fileId=id, mimeType="text/plain")
+        file = self._service().files().get(fileId=id, supportsAllDrives=True).execute()
```
Comment (hwchase17) [inline langchain/document_loaders/googledrive.py:177]: can we do `service = self._service()` and then use in both these lines?

Comment (Yvelo) [reply langchain/document_loaders/googledrive.py]: see above

**Comment by Yvelo** [comment]
II have further improved this Pool request to add support to Slides and allow load_file_from_ids to check for the appropriate loader.

**Comment by eyurtsev** [comment]
cross-referencing: https://github.com/hwchase17/langchain/pull/5135

**Comment by pprados** [comment]
The pull request [5135](https://github.com/hwchase17/langchain/pull/5135) resolve this.

**Comment by leo-gan** [comment]
#9999 resolves it.

**Pull request closed without merging**